patches/server/0969-Fix-spawn-animals-and-spawn-monsters-settings.patch

```diff
@@ -0,0 +1,73 @@
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Wed, 21 Dec 2022 10:38:41 -0800
+Subject: [PATCH] Fix spawn-animals and spawn-monsters settings
+
+
+diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
+index fbe209a66c77c47935ad026dd3e45e682af91fd8..974c25eb49cf4b14fc5b30b894af68fd60bed65b 100644
+--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
++++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
+@@ -725,18 +725,12 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
+         if (!nbt.isEmpty()) {
+             // CraftBukkit start - these are spawned serialized (DefinedStructure) and we don't call an add event below at the moment due to ordering complexities
+             world.addWorldGenChunkEntities(EntityType.loadEntitiesRecursive(nbt, world).filter((entity) -> {
+-                boolean needsRemoval = false;
+-                net.minecraft.server.dedicated.DedicatedServer server = world.getCraftServer().getServer();
+-                if (!server.areNpcsEnabled() && entity instanceof net.minecraft.world.entity.npc.Npc) {
++                if (world.shouldDiscardEntityInWorld(entity)) { // Paper - this needs to be per-world & simplified
+                     entity.discard();
+-                    needsRemoval = true;
+-                }
+-                if (!server.isSpawningAnimals() && (entity instanceof net.minecraft.world.entity.animal.Animal || entity instanceof net.minecraft.world.entity.animal.WaterAnimal)) {
+-                    entity.discard();
+-                    needsRemoval = true;
++                    return false; // Paper
+                 }
+                 checkDupeUUID(world, entity); // Paper
+-                return !needsRemoval;
++                return true; // Paper
+             }));
+             // CraftBukkit end
+         }
+diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
+index 5a5ff40df37db9cbd53c584ed26a3ce4888b29c0..c1811c3ab960f8587a3630b96bbad4523c436042 100644
+--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
++++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
+@@ -784,6 +784,11 @@ public class ServerLevel extends Level implements WorldGenLevel {
+     private boolean shouldDiscardEntity(Entity entity) {
+         return !this.server.isSpawningAnimals() && (entity instanceof Animal || entity instanceof WaterAnimal) ? true : !this.server.areNpcsEnabled() && entity instanceof Npc;
+     }
++    // Paper start - per world version of above method
++    public boolean shouldDiscardEntityInWorld(Entity entity) {
++        return !this.getChunkSource().spawnFriendlies && (entity instanceof Animal || entity instanceof WaterAnimal) ? true : !this.server.areNpcsEnabled() && entity instanceof Npc; // TODO per-world npc
```

Comment:
This seems like it could also be done like this, but good either way.

```suggestion
+        return (!this.getChunkSource().spawnFriendlies && (entity instanceof Animal || entity instanceof WaterAnimal)) || (!this.server.areNpcsEnabled() && entity instanceof Npc); // TODO per-world npc
```

Reply:
I did it that way to copy more exactly what was done above since its copied directly from there. I think, for that reason, its fine to leave as is